### DISCUSSION TITLE
Proxmox supported versions (and whether to deprecate/drop them)

### DISCUSSION THREAD

**Discussion by Lithimlin** [ansible-collections/news-for-maintainers#73] [Discussions on announcements]
To work on some issues, I am currently refactoring the [`proxmox` module](https://docs.ansible.com/ansible/latest/collections/community/general/proxmox_module.html).

While working on that, it came up again and again that technically, any version of Proxmox, even below 4.0 is supported. This raised the question for me on whether this is 1) necessary or 2) even desirable. The Debian versions those old Proxmox versions are based on are long at their EOL and get no more (security) updates.

This particularly came up when trying to figure out how the `storage` parameter works now.
In the module docs, it says for the [`disk` parameter](https://docs.ansible.com/ansible/latest/collections/community/general/proxmox_module.html#parameter-disk):

> Older versions of Proxmox will accept a numeric value for size using the storage parameter to automatically choose which storage to allocate from [...]

And while I *can* find the `storage` parameter in the [API Docs](https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/lxc) (see the `POST` tab), I cannot figure out if/for what it is even still being used now. This is not necessarily a problem regarding supported versions, but further complicates the existing issues.

To summarize, my main questions are:

1. Is support of *all* Proxmox versions necessary?
2. Is support of *all* Proxmox versions desirable?
3. Does anyone know where I can check what the `storage` parameter is being used for since the numeric value for `disk`/`rootfs` is no longer valid?
4. Additionally, does anyone know when this behaviour changed? I could narrow it down to somewhere between versions 4.0 and 5.0 using the Proxmox Wiki ([old post](https://pve.proxmox.com/mediawiki/index.php?title=Linux_Container&direction=prev&oldid=9026) vs [new post](https://pve.proxmox.com/mediawiki/index.php?title=Linux_Container&direction=next&oldid=9026)).

**Comment by mariolenz** [comment]
I'm not sure if this is the best place for a question you have on a specific module in a specific collection.

I suggest to ask directly [in the collection](https://github.com/ansible-collections/community.general/issues), on [Matrix](https://matrix.to/#/#community:ansible.com) or in the [forum](https://forum.ansible.com/).

**Reply by Lithimlin** [reply]
I'd posted an issue in the collection before, but found an announcement there that said to post discussions here.

**Reply by Lithimlin** [reply]
I'll check out the Forum later and maybe just post it to Matrix, though it would probably get lost there fairly quickly.

**Reply by felixfontein** [reply]
> but found an announcement there that said to post discussions here.

Can you please link to that announcement? That should be taken down ASAP.

The right place for such a discussion is the Forum, since there are no active maintainers for the proxmox modules and it's hard to get any opinion from other users or (potential) contributors in the community.general repository.

**Reply by mariolenz** [reply]
> I'd posted an issue in the collection before, but found an announcement there that said to post discussions here.

ansible-collections/community.general#8407 and ansible-collections/community.general#9199? I don't find any comment in those issues that redirects you here. I might be wrong, though, because I've only had a _very_ quick look.

Could you please add a link to the announcement you've mentioned?

**Reply by Lithimlin** [reply]
https://github.com/ansible-collections/community.general/issues/4087

I might have misunderstood the post though then.  It says to not hold any (unnecessary) discussions in issues and I took that to mean that discussions should not be issues. I'll reopen my original issue then.

**Reply by felixfontein** [reply]
That sentence was meant for discussions about announcements made in news-for-maintainers. I closed that issue since it's pretty outdated anyway.
